### PR TITLE
pfappserver updates

### PR DESCRIPTION
**Pull request by satkunas**
# Description
Continue work on the pfappserver

# Impacts
Increased PR count :100: 

# Delete branch after merge
YES

### CODE REVIEW

File: html/pfappserver/root/static.alt/src/components/pfSearch.vue [hunk 1]
```diff
@@ -26,22 +26,22 @@
           </b-button-group>
         </b-container>
       </b-form>
-      <b-modal v-model="showExportJsonModal" size="lg" centered id="nodeExportJsonModal" :title="$t('Export to JSON')">
-        <b-form-textarea ref="nodeExportJsonTextarea" v-model="JSON.stringify(condition)" :rows="3" :max-rows="3" readonly></b-form-textarea>
+      <b-modal v-model="showExportJsonModal" size="lg" centered id="exportJsonModal" :title="$t('Export to JSON')">
+        <b-form-textarea ref="exportJsonTextarea" v-model="JSON.stringify(condition)" :rows="3" :max-rows="3" readonly></b-form-textarea>
```
Comment (cgx) [inline html/pfappserver/root/static.alt/src/components/pfSearch.vue]: `v-model` should be a variable, not the result of a function call.

File: html/pfappserver/root/static.alt/src/views/Nodes/_components/NodeView.vue [hunk 2]
```diff
@@ -3,12 +3,15 @@
   <b-form @submit.prevent="save()">
     <b-card no-body>
       <b-card-header>
-        <b-button-close @click="close"><icon name="times"></icon></b-button-close>
+        <b-button-close @click="close" v-b-tooltip.hover.left :title="$t('Close [ESC]')"><icon name="times"></icon></b-button-close>
```
Comment (cgx) [inline html/pfappserver/root/static.alt/src/views/Nodes/_components/NodeView.vue]: I propose an [initial delay](https://bootstrap-vue.js.org/docs/directives/tooltip/#directive-syntax-and-usage) for the tooltip:

` v-b-tooltip.hover.left.d300 `

File: html/pfappserver/root/static.alt/src/views/Nodes/_components/NodeView.vue [hunk 3]
```diff
@@ -46,6 +49,9 @@
         </b-tab>
 
         <b-tab title="Fingerbank">
+          <template slot="title">
+            {{ $t('Fingerbank') }}
+          </template>
```
Comment (cgx) [inline html/pfappserver/root/static.alt/src/views/Nodes/_components/NodeView.vue]: Fingerbank will (hopefully) never be translated 😉 Remove the slot and simply keep the title attribute.

File: html/pfappserver/root/static.alt/src/views/Nodes/_components/NodeView.vue [hunk 4]
```diff
@@ -254,6 +287,9 @@ export default {
     }
   },
   methods: {
+    onDelete (event) {
+      console.log(['onDelete', event])
+    },
```
Comment (cgx) [inline html/pfappserver/root/static.alt/src/views/Nodes/_components/NodeView.vue]: To remove?

File: html/pfappserver/root/static.alt/src/components/pfMixinSavedSearch.vue [hunk 5]
```diff
@@ -0,0 +1,47 @@
...
+    deleteSavedSearch (search) {
+      this.$store.dispatch(`$_${this.storeName}/deleteSavedSearch`, search)
+    },
+    routeSavedSearch (search) {
+      return { name: this.routeName, query: { query: JSON.stringify(search.query) } }
+    }
+  },
+  created () {
+    // Called before the component's created function.
+    if (!this.$options.props.storeName) {
```
Comment (cgx) [inline html/pfappserver/root/static.alt/src/components/pfMixinSavedSearch.vue]: You could simply [make the prop required](https://vuejs.org/v2/guide/components-props.html#Prop-Validation).

File: html/pfappserver/root/static.alt/src/components/pfMixinSavedSearch.vue [hunk 6]
```diff
@@ -0,0 +1,47 @@
...
+      throw new Error(`Missing 'props.storeName' in properties of component ${this.$options.name}`)
+    }
+    if (!this.$options.props.routeName) {
```
Comment (cgx) [inline html/pfappserver/root/static.alt/src/components/pfMixinSavedSearch.vue]: You could simply [make the prop required](https://vuejs.org/v2/guide/components-props.html#Prop-Validation).

File: html/pfappserver/root/static.alt/src/components/pfSearch.vue [hunk 7]
```diff
@@ -171,12 +175,18 @@ export default {
...
-      this.$store.dispatch('$_nodes/addSavedSearch', {name: this.saveSearchString, query: this.condition}).then(response => {
+      this.$store.dispatch(`$_${this.storeName}/addSavedSearch`, {name: this.saveSearchString, query: this.condition}).then(response => {
         _this.saveSearchString = ''
         _this.showSaveSearchModal = false
       })
     }
   },
+  created () {
+    // Called before the component's created function.
+    if (!this.$options.props.storeName) {
```
Comment (cgx) [inline html/pfappserver/root/static.alt/src/components/pfSearch.vue]: You could simply [make the prop required](https://vuejs.org/v2/guide/components-props.html#Prop-Validation).

**Review by cgx — DISMISSED** [review]

File: html/pfappserver/root/static.alt/src/views/Nodes/_components/NodeView.vue (hunk 4 above)
Comment (satkunas) [reply html/pfappserver/root/static.alt/src/views/Nodes/_components/NodeView.vue]: To be continued....

**Comment by satkunas** [comment]
@cgx all up-to-date.

Merge when ready.

**Pull request merged**